app/controllers/application_controller.rb

```diff
@@ -1,4 +1,9 @@
 # frozen_string_literal: true
 
 class ApplicationController < ActionController::API
+  before_action do
+    # This is to access S3 URLs with #service_url, which returns a url that expires in 5 Minutes
+    # In contrast, url_for returns permanent URL, which for privacy/security reasons, we do not want.
+    ActiveStorage::Current.host = request.base_url
+  end
```

Comment:
Could this be moved to `ApiController` instead, and perhaps made part of a dedicated method rather than including the code within a block pls?

Reply:
yes we can, `ApiController` seemed already growing a lot, but happy to move it there, np.

Reply:
done